c2corg_ui/static/js/search/searchfilters.js

```diff
@@ -189,4 +223,42 @@ app.SearchFiltersController.prototype.selectOption = function(prop, val, event)
 };
 
 
+/**
+ * @export
+ */
+app.SearchFiltersController.prototype.clear = function() {
+  for (var key in this.filters) {
+    this.location_.deleteParam(key);
+  }
+  this.filters = {};
+  this.orientations = [];
+  this.scope_.$root.$emit('searchFilterClear');
+};
+
+
+/**
+ * @param {string} orientation
+ * @param {app.SearchFiltersController} ctrl
+ * @param {goog.events.Event | jQuery.Event} e
+ * @param {string} filterName Name of the filter param in the URL.
+ * @export
+ */
+app.SearchFiltersController.prototype.setOrientation = function(orientation,
+    ctrl, e, filterName) {
+  if (this.orientations.indexOf(orientation) === -1) {
+    this.orientations.push(orientation);
+  } else {
+    this.orientations = this.orientations.filter(function(val) {
+      return val !== orientation;
+    });
```

Comment:
When using this ``setOrientation`` method (triggered when clicking on one of the directions of the SVG compass, eg. at https://github.com/c2corg/v6_ui/blob/advanced-search-step2/c2corg_ui/templates/helpers/orientations.svg?short_path=31b8504#L62):
* either the orientation is not already in the orientations list and we push it into the list
* or it is already in the list and we remove it using this "filter" call that make sure that all instances (in case several were - by mistake? - available) of this orientation are removed (we keep only the values that are not equal to the orientation we want to remoev).